Add sphinx based docs with rtd theme for ab

kmehant opened the pull request:
Shifted ansible-bender technical documentation to read-the-docs

kmehant commented:
@TomasTomecek We have got a good number of rst files to maintain and change for new modifications.
What I was thinking is to divide the present `README.md` to similar chucks as I have done for this documentation part and write a pipeline that can reflect the changes made to the markdown files into corresponding rst files and keep the documentation up to date. This way the present README can also be slimmed down and looks more welcoming also. What are your thoughts regarding this?

TomasTomecek commented:
wow this is nice, thank you! it will take me some time to get through this

+1 on shortening the present README and transform it into neat docs

kmehant commented:
@TomasTomecek I have done that way, please have a look into it :)

TomasTomecek commented on README.md:
```diff
@@ -35,384 +35,14 @@ while ab takes care of image builds only.
+* [Contribution guide](https://github.com/ansible-community/ansible-bender/docs/md_docs/contributing.md)
 
 
 ### Ansible-bender in OKD
```
please move this to docs as well

kmehant replied:
Moving README.md to docs will be a problem as documentation preview on GitHub will not be shown, as GitHub looks for README.md in the root of the project. 
What shall we do then?

TomasTomecek replied:
I meant to move the "AB in OKD" section to docs :)

TomasTomecek commented on README.md:
```diff
@@ -35,384 +35,14 @@ while ab takes care of image builds only.
-add a tag `stop-layering` to a task which will disable layering (and caching)
-for that task and all the following ones.
+## Contents
+* [Features](https://github.com/ansible-community/ansible-bender/docs/md_docs/features.md)
```
I'd probably leave features in README so people can actually get interested in the project by the first look.

TomasTomecek replied:
In future, we can describe futures in detail in the documentation.

TomasTomecek commented on docs/md_docs/contributing.md:
```diff
@@ -0,0 +1,72 @@
+# Contributing to ansible-bender
```
would be awesome to merge this with CONTRIBUTING.md so that we don't have two distinct contributing guides

kmehant commented:
> I had to install python3-m2r for sphinx to convert markdown to rest
> 
> but sadly, couldn't get html files :/
> 
> ```
> Running Sphinx v1.8.4
> building [mo]: targets for 0 po files that are out of date
> building [html]: targets for 8 source files that are out of date
> updating environment: 8 added, 0 changed, 0 removed
> reading sources... [100%] usage                                                                                                                                              
> 
> Sphinx error:
> master file /home/tt/g/TomasTomecek/ansible-bender/docs/source/contents.rst not found
> make: *** [Makefile:20: html] Error 2
> ```

@TomasTomecek I have used m2r package, which I have added to `conf.py` as an extension. I have updated requirements.txt.
I have added features into index and README directly. I was able to build the html files 🤔

kmehant commented on docs/requirements.txt:
```diff
@@ -0,0 +1,27 @@
+idna==2.8
+imagesize==1.2.0
+Jinja2==2.10.3
+m2r==0.2.1
```
This is the package I have used for conversion

TomasTomecek replied:
Nice, works just fine now!

I need to allocate some time to merge this and set up RTD for bender. Since it's https://www.devconf.info/cz/ happening this week, it will take me some time to do that. Mehant, unless you want set up RTD yourself :)

thrix commented:
/packit test

kmehant commented:
@TomasTomecek I have made all the requested changes :)
I haven't setup RTD earlier for production software, so I wish you do it :)

TomasTomecek commented:
Thanks, Mehant. After three months I finally found time to merge this and put out. Better late than never. :)
